### ISSUE TITLE
Add ability to trigger start and stop though other field(s)

### ISSUE THREAD

**Issue by davwillev** [davwillev/stopwatch#52]
This came up several times in Q&A at the consortium presentation. Might this be related to #8 (Interface for controlling multiple stopwatches)?

**Comment by grezniczek** [comment]
I think this was more related to timing a survey, i.e. how long from start to finish.

There really is no reliable way to time survey duration. REDCap can do this already, but it all depends on compliance of the survey participant. If they decide to take a nap while filling out a survey despite instructions saying they should try to do it in one uninterrupted session, then a stopwatch will not help either ...

UNLESS ... it would autostart (and autostop) and work as an incentive for people ... hmm ....
But in my opinion this should be a separate EM.

**Comment by davwillev** [comment]
I agree that participant behaviour will inevitably affect this, but the researchers might still want to know this?

Also, having ‘remote’ triggers seems like a nice option, but admittedly I can’t think of a realistic use case

**Comment by davwillev** [comment]
I have thought of one use case for remote triggering:
If a researcher is completing a form, it can be useful to record how long one or more stages of data collection are taking. Certainly in the UK (and preumably elsewhere), the cost of data collection from NHS pesonnel is based on such information.

**Comment by grezniczek** [comment]
This is an interesting use case.
I have been thinking about something like this today ... How about letting a stopwatch run invisibly in the background, starting when a form or survey loads. It'll be a lap stopwatch. Each time the focus moves to a form element, a lap is taken, including the name of that form field. The timer ends with submission. Multiple forms / surveys can be captured to the same repeating form. This would allow detailed analysis of the "movements" of the user on the form, if they went in order, how much time they spent between clicks, etc. 
Sounds quite powerful, and most of the groundwork for such functionality now already exists with the lap stopwatch :)

**Comment by davwillev** [comment]
As always, your implementation ideas are excellent @grezniczek !
Yes, this would indeed be a powerful adaptation of the stopwatch functionality. I believe that this would be very popular with REDCap users :)

In case we think of later use cases for remote triggering, do you want to leave the visibility of the triggered stopwatch as an option (default = invisible), and likewise the timer ending as an option (default = form submission)?

**Comment by davwillev** [comment]
Oh, and also the timer starting as an option (default = form loading)?

**Comment by davwillev** [comment]
I think I have thought up another use case for remote triggering and the 'time-tracking' function that you described. In the psychological and rehabilitation literature there are several timed reaction/evaluation tests that require individual and itemised time captures. The most known of these is [the Stroop test](https://en.wikipedia.org/wiki/Stroop_effect), but I am also aware of image evaluation tests that are also timed (discrimination of left or right side of the body in chronic pain or neurological patients).

I'm not sure whether a REDCap field can be made to immediately appear and disappear on the screen without scrolling (which would adversely affect the assessment), but assuming this can be overcome, the lap stopwatch should do the job here.

**Comment by grezniczek** [comment]
Just for reference: https://community.projectredcap.org/questions/20373/survey-start-time-status.html?childToView=86204#comment-86204

**Comment by galambda** [comment]
Hello everyone!

I have started to get to know the stopwatch and I am impressed with the range of customization and functionality! I've seen that you started discussing it here in the thread but is it possible to start the stopwatch in the background during a survey and stop when the submission button is pressed?

Thanks and bests,
Philipp

**Comment by galambda** [comment]
> 
> 
> Hello everyone!
> 
> I have started to get to know the stopwatch and I am impressed with the range of customization and functionality! I've seen that you started discussing it here in the thread but is it possible to start the stopwatch in the background during a survey and stop when the submission button is pressed?
> 
> Thanks and bests,
> Philipp

Hello again,
I hope I interpret the thumb-up correctly as a positive response. Can you hence please explain how to do it in REDCap as I could not find it in the provided documentation.

Thanks,
Philipp

**Comment by davwillev** [comment]
Hi @galambda 
My ambiguous thumbs up was to welcome your favourable comments towards the module. I agree that this will be a useful addition to the stopwatch functionality though.

Unfortunately, this functionality has not yet been added.
@grezniczek is the architect of the module, so would need to add this when he has time. I would do the honours but he left my coding skills standing a long time ago :)

Best wishes

**Comment by galambda** [comment]
> 
> 
> Hi @galambda
> My ambiguous thumbs up was to welcome your favourable comments towards the module. I agree that this will be a useful addition to the stopwatch functionality though.
> 
> Unfortunately, this functionality has not yet been added.
> @grezniczek is the architect of the module, so would need to add this when he has time. I would do the honours but he left my coding skills standing a long time ago :)
> 
> Best wishes

I see - thank you very much for the clarification :)
I'll keep my fingers crossed and will check by from time to time. Until then I'll have to go back to the default REDCap solution.

Cheers!

**Comment by grezniczek** [comment]
@galambda @davwillev 
Thanks for the input and response. Survey duration is a tricky issue and as long as you cannot control (which you never can) what people taking a survey do while taking it, it is quite meaningless as a measure. At best, it can give you an idea about the _minimum_ time it takes to complete a survey (and this only when there are sufficient internal controls to exclude "click-through" survey takers).

Due to this, a "start" to "finish" functionality is not useful - and not needed, as REDCap already registers the time a survey is submitted (not sure from the top of my head if the start time is also recorded) and the start time can be captured with one of the  @NOW action tags coupled with @HIDDEN-SURVEY. If your survey has multiple pages, you can use multiple @NOW action tags (one per section).

Another issue is starting the timer with another action on a form, such as choosing a radio option or a checkbox. But I have not heard any concrete use case for this yet.
